pkg/isbsvc/jetstream_service.go

```diff
@@ -316,9 +315,9 @@ func (jss *jetStreamSvc) GetBufferInfo(ctx context.Context, buffer string) (*Buf
 	return bufferInfo, nil
 }
 
-// CreateUXWatermarkFetcher is used to create watermark fetcher for the given bucket.
-func (jss *jetStreamSvc) CreateUXWatermarkFetcher(ctx context.Context, bucketName string, fromBufferPartitionCount int, isReduce bool) ([]fetch.UXFetcher, error) {
-	var watermarkFetchers []fetch.UXFetcher
+// CreateProcessorManagers is used to create watermark fetcher for the given bucket.
+func (jss *jetStreamSvc) CreateProcessorManagers(ctx context.Context, bucketName string, fromBufferPartitionCount int, isReduce bool) ([]*processor.ProcessorManager, error) {
```

Comment:
Would it be better if we separate this interface into two `CreateWatermarkStores()` and `CreateProcessorManagers`? From the ownership perspective, the processor managers don't own the stores, hence the store creation is better to be independent of the processor manager's creation.

Reply:
you meant `CreateWatermarkWatchers()` ?...